components/brave_wallet_ui/components/buy-with-button/index.tsx

```diff
@@ -0,0 +1,72 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+
+import {
+  StyledWrapper,
+  Button,
+  CaratDown,
+  Dropdown,
+  Option
+} from './style'
+import { BraveWallet, BuyOption } from '../../constants/types'
+import { RampLogo, WyreLogo } from '../buy-send-swap/buy/style'
+
+export interface Props {
+  children?: React.ReactNode
+  options: BuyOption[]
+  value: BraveWallet.OnRampProvider
+  closeOnSelect?: boolean
+  disabled?: boolean
+  onSelect: (value: BraveWallet.OnRampProvider) => void
+}
+
+const SelectBuy = (props: Props) => {
+  const { children, options, value, closeOnSelect, disabled, onSelect } = props
+  const [isOpen, setIsOpen] = React.useState(false)
+
+  const onClick = () => {
+    setIsOpen(!isOpen)
```

Comment:
`setIsOpen(prevIsOpen => !prevIsOpen)`, otherwise, you need useCallback with isOpen as a dependency